routes/api/oauth.ts

```diff
@@ -0,0 +1,29 @@
+import type { Handlers } from "$fresh/server.ts";
+import { assert } from "std/testing/asserts.ts";
+import { createSupabaseClient } from "@/utils/supabase.ts";
+import type { Provider } from "@supabase/supabase-js";
+
+export const handler: Handlers = {
+  async POST(request) {
+    const form = await request.formData();
+    const provider = form.get("provider");
+
+    assert(typeof provider === "string");
```

Comment:
I think this should return `400` response instead of throwing

Reply:
Done.